[ulricolo7] iP

ulricolo7 opened the pull request:
# Neonbot
> “Burgers.” – What you want when you are hungry.

Neonbot frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ulricolo7 committed:
Level-0

ulricolo7 committed:
Level-1

ulricolo7 committed:
Level-2

ulricolo7 committed:
Level-3

ulricolo7 committed:
Level-4

ulricolo7 committed:
A-TextUiTesting

ulricolo7 committed:
Level-5

ulricolo7 committed:
Level-6, A-Collection

ulricolo7 committed:
Level-7

ulricolo7 committed:
Level-8

jktang14 commented on src/main/java/Neon.java:
```diff
@@ -0,0 +1,244 @@
+public class Neon {
+    private static final String NAME = "Neon";
+    private static final String DASH_BREAK = "-----------------------------------";
+    private static ArrayList<Task> list = new ArrayList<>();
```
Perhaps could change the naming to specify what type of list this is? (Ie. taskList)

jktang14 commented on src/main/java/Neon.java:
```diff
@@ -0,0 +1,244 @@
+    private static ArrayList<Task> list = new ArrayList<>();
+    private static int lastListIndex = 0;
+
+    private static void greetingLine() {
```
Perhaps you could rephrase this to sound more like a verb, for example `printGreetingLine()`

jktang14 commented on src/main/java/Neon.java:
```diff
@@ -0,0 +1,244 @@
+            String line = scanner.nextLine();
+            String[] content = line.split("/");
+
+            boolean completed;
```
Perhaps you could change this naming to sound more like a boolean, Eg. isCompleted

jktang14 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,26 @@
+public class Event extends Task {
+    private String startDT;
```
Perhaps you could change this naming to make it clearer what DT means

jktang14 reviewed (commented):
LGTM. Most methods, variable names follow the code naming guidelines, however there are small instances of violations, correcting these should improve your code quality as a whole 👍

huangjiaxi1111 commented on src/main/java/Neon.java:
```diff
@@ -0,0 +1,244 @@
+        String first = ansObj.next();
+
+        switch (first) {
+            case "todo":
```
Should case be indented?

huangjiaxi1111 commented on src/main/java/Neon.java:
```diff
@@ -0,0 +1,244 @@
+        System.out.println(DASH_BREAK);
+    }
+
+    private static void closingLine() {
```
Just to add on, here can change to `printClosingLine()`

huangjiaxi1111 commented on src/main/java/Neon.java:
```diff
@@ -0,0 +1,244 @@
+    private static int lastListIndex = 0;
+
+    private static void greetingLine() {
+        System.out.println(DASH_BREAK);
```
Maybe you can write a `printDashBreak()` to abstract this out. This can make your code cleaner!

huangjiaxi1111 reviewed (commented):
LGTM! Your code managed to follow most part of the coding standard.
I like how you wrapped your code to make sure that each line is not too long and used correct formats for variable, method, and class naming.
To improve, maybe you can include header comments and put classes into packages.

ulricolo7 committed:
A-MoreOOP

ulricolo7 committed:
A-Packages

ulricolo7 committed:
Merge branch 'add-gradle-support'

ulricolo7 committed:
A-Gradle, A-JUnit

ulricolo7 committed:
Update build.gradle

ulricolo7 committed:
Formatting updates

ulricolo7 committed:
A-JavaDoc

ulricolo7 committed:
Level-9

ulricolo7 committed:
Merge branch 'branch-Level-9'

ulricolo7 committed:
Add GUI

ulricolo7 committed:
Add GUI

ulricolo7 committed:
Update GUI

ulricolo7 committed:
Update Neon.java

ulricolo7 committed:
Add Assertions

Add Assertion into the bot

All of the assertions added are within Parser.java. The intention is to test and check for valid inputs during testing.

Edited much of the code to reflect the intended interaction better as well.

ulricolo7 committed:
Merge pull request #2 from ulricolo7/branch-A-Assertions

Add Assertions

ulricolo7 committed:
Fix code quality

This commit is made to fix a few potential structural issues with the code and also ensure the overall quality of the code.

ulricolo7 committed:
Merge pull request #3 from ulricolo7/branch-A-CodeQuality

Fix code quality

ulricolo7 committed:
Add Undo

ulricolo7 committed:
Update README.md

ulricolo7 committed:
Create Ui.png

ulricolo7 committed:
Update README.md
